contracts/token/PoolFDT.sol

```diff
@@ -1,30 +1,58 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 pragma solidity 0.6.11;
 
+import "lib/openzeppelin-contracts/contracts/token/ERC20/SafeERC20.sol";
+
 import "./ExtendedFDT.sol";
+import "./ERC2258.sol";
 
 /// @title PoolFDT inherits ExtendedFDT and accounts for gains/losses for Liquidity Providers.
-abstract contract PoolFDT is ExtendedFDT {
+abstract contract PoolFDT is ExtendedFDT, ERC2258 {
+    
     using SafeMath       for uint256;
     using SafeMathUint   for uint256;
     using SignedSafeMath for  int256;
     using SafeMathInt    for  int256;
+    using SafeERC20      for  IERC20;
+
+    IERC20 public fundsToken;
```

Comment:
Cannot be immutable as it is not known a construction-time. See `Pool` comment for more.